### PR TITLE
Add new command GetSerialInterfaces and SetSerialInterface

### LINKED ISSUE #66142: redfish_info / redfish_config - Manager - GetSerialInterfaces / SetSerialInterface
##### SUMMARY

This feature would implement a GetSerialInterfaces command for the Manager category of redfish_info, to retrieve all serial interfaces and their properties, like BitRate. And also implement a SetSerialInterface command for the Manager category of redfish_config, to set SerialInterface's properties, like enable/disable the interface, change BitRate.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

redfish_info.py
redfish_config.py
redfish_utils.py

##### ADDITIONAL INFORMATION

These 2 commands would help user to get serial interfaces' info or change serial interface.

```yaml
---
- hosts: localhost
  gather_facts: false
  vars:
    baseuri: 10.245.39.251
    username: USERID
    password: PASSW0RD
  tasks:
    - name: Get Serial Interfaces information
      redfish_info:
        category: Manager
        command: GetSerialInterfaces
        baseuri: "{{ baseuri }}"
        username: "{{ username }}"
        password: "{{ password }}"
    - name: Set Serial Interface
      redfish_config:
        category: Manager
        command: SetSerialInterface
        baseuri: "{{ baseuri }}"
        username: "{{ username }}"
        password: "{{ password }}"
        serial_interface_id: "1"
        serial_interface_config:
          InterfaceEnabled: True
          BitRate: 115200
```

### PR DESCRIPTION
**Pull request by renxulei**
##### SUMMARY

Add GetSerialInterfaces command to get all serial interfaces info of Manager, and add SetSerialInterface command to change the setting of serial interface.

Fixes #66142 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redfish_info.py
redfish_config.py
redfish_utils.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/redfish_utils.py
Comment (billdodd) [inline lib/ansible/module_utils/redfish_utils.py]: For this case (no id specified; use the first one), it looks like there is some code missing to read the data from the first URI and then set `target_serial_interface_current_setting` to the data.

**Review by billdodd — COMMENTED** [review]
Thanks for the PR. Looks quite good. But please see the one comment below.

File: lib/ansible/module_utils/redfish_utils.py
Comment (renxulei) [reply lib/ansible/module_utils/redfish_utils.py]: Many thanks, Bill, good catch!  I have modified the code, please help to review again, thanks!

**Comment by billdodd** [comment]
Looks good. Thanks.

shipit

**Comment by renxulei** [comment]
@mraineri @tomasg2012 @jose-delarosa  Could you help to review this? Thanks!

**Comment by tomasg2012** [comment]
shipit

**Review by panyy3 — APPROVED** [review]

**Comment by renxulei** [comment]
@billdodd @tomasg2012  thank you all to review this PR and gave "shipit" label, but this PR is requested 6 "approving reviews" before it is merged. Could you help to give your "approving reviews" for this PR?  Thanks!

**Comment by billdodd** [comment]
@renxulei I checked with the core team and the "At least 6 approving reviews are required" message is a temporary condition while the devel branch is frozen. (It is currently frozen for a migration operation.) Things are expected to get back to normal around the end of the month. No additional approvals are required.

**Comment by renxulei** [comment]
@billdodd Got you, many thanks for your quick response and confirmation!

**Review by rajeevkallur — APPROVED** [review]
Looks good!

**Comment by renxulei** [comment]
@billdodd Hi, Bill, I am trying to rebase my code, but I found there are no any modules of Redfish in devel branch of ansible project, it seems there is big change in ansible's code structure.  Do you know what happened and how can I add the commands related with Redfish?  Thanks for your help!

**Comment by billdodd** [comment]
@renxulei Yes, as you have seen, the Redfish modules are no longer in the ansible/ansible repository in the devel branch. This is a part of a restructuring for the Ansible 2.10 release that will separate the Ansible core from the bulk of the non-core modules and plugins. The non-core modules and plugins (like Redfish and others) will still be part of Ansible but will reside in other repositories. You can read more about this here:

 https://github.com/ansible-collections/overview/blob/master/README.rst

For Ansible 2.10, the Redish modules will reside here:
https://github.com/ansible-collections/community.general

So, you will not be able to rebase your PR here in the ansible/ansible repo. When the community.general repo is ready to accept PRs, you will need to resubmit your PR there. When will that be? I'm not sure, but you can watch for updates in the [Timeline section](https://github.com/ansible-collections/overview/blob/master/README.rst#timeline) of the `README.rst` file I referenced above. See the line that mentions "community.general accepts new Pull Requests":

> - TBC, community.general accepts new Pull Requests (PRs).

Once they start accepting PRs, that line should be updated with the date. Hopefully, that will be soon.

**Comment by renxulei** [comment]
@billdodd  Bill, thank you very much for your detailed explanation, very helpful. I will wait community.general to accept PR then see how to contribute the code.

**Comment by billdodd** [comment]
@renxulei FYI - Here is a link with some information about how to get this PR migrated over to the new ansible-collections/community.general repository:

https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md

I just did this recently with my open Redfish issue and PR. I did the migration manually by recreating the issue and the PR over in the new repository. I did not try to use the prmove script referenced in the link above. (It looked more complicated than just manually recreating.)
For reference, if you want to look at the issue and PR I migrated, here are the links:

Original issue: #58941 
Original PR: #68053 

New issue: https://github.com/ansible-collections/community.general/issues/493
New PR: https://github.com/ansible-collections/community.general/pull/494

**Comment by renxulei** [comment]
@billdodd Thank you very much! This is really helpful! I will follow your way to migrate the issue and PR.

**Pull request closed without merging**
